### PR TITLE
Allow target.requires_grad in l1_loss and mse_loss

### PR DESCRIPTION
**Pull request by szagoruyko**
This bug cost me a lot of time in several projects, 0.2 would silently accept target.requires_grad and would backprop zeroes in `l1_loss` and `mse_loss`. It's fixed in master already (with exception), and this PR allows target.requires_grad by explicitly defining the loss functions.

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -1267,6 +1267,16 @@ def binary_cross_entropy_with_logits(input, target, weight=None, size_average=Tr
         return loss.sum()
 
 
+def _poinwise_loss(lambd, lambd_optimized, input, target, size_average=True, reduce=True):
```
Comment (apaszke) [inline torch/nn/functional.py]: I think you wanted to make it _poin**t**wise_loss

**Review by apaszke — COMMENTED** [review]
Looks good, but please fix the typo

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
linter still has a few complaints
